snuba/datasets/configuration/generate_config_docs.py

```diff
@@ -0,0 +1,31 @@
+from tempfile import TemporaryDirectory
+
+import jsonschema2md
+
+from snuba import settings
+from snuba.datasets.configuration.json_schema import V1_ALL_SCHEMAS
+
+CONFIGURATION_DOCS_PATH = f"{settings.ROOT_REPO_PATH}/docs/source/configuration"
+
+
+if __name__ == "__main__":
```

Comment:
If our JSON schema was not embedded in python but present in files that can parse raw JSON schema text, would we be able to use an existing utility rather than rolling our own here?

Not trying to stop you from merging this but I want to continue to reiterate that I think there are benefits to not having the JSON schema be encoded in Python dicts (e.g. you could have [first-class text editor support](https://code.visualstudio.com/docs/languages/json#_mapping-to-a-schema-in-the-workspace) for authoring new datasets, or validate with other [language bindings](https://json-schema.org/implementations.html#validators) if we want to support Rust in the future)

Reply:
I agree. The work behind implementing jsonschema validation for pre-commit further supports this idea. Currently, our pre-commit uses a custom script (https://github.com/getsentry/snuba/blob/master/snuba/validate_configs.py) to validate our configs. This was required because the built-in hook`check-jsonschema` (https://github.com/python-jsonschema/check-jsonschema) only accepts .json or .yaml files as an input parameter.